core/src/main/java/io/questdb/griffin/engine/functions/SymbolFunction.java

```diff
@@ -176,11 +176,14 @@ public final int getStrLen(Record rec) {
     @Override
     public final long getTimestamp(Record rec) {
         final CharSequence value = getSymbol(rec);
-        try {
-            return IntervalUtils.parseFloorPartialTimestamp(value);
-        } catch (NumericException e) {
-            throw CairoException.nonCritical().position(0).put("invalid timestamp: [").put(value).put(']');
+        if (value != null) {
+            try {
+                return IntervalUtils.parseFloorPartialTimestamp(value);
+            } catch (NumericException e) {
+                throw CairoException.nonCritical().position(0).put("invalid timestamp: [").put(value).put(']');
```

Comment:
It's a higher-level, abstract class and in those, we usually don't bother accepting positions in the constructor. E.g. `SymbolConstant` inherits this class and since we reuse constant instances, it's tricky to return the correct position from here.

Reply:
With `SqlException` we use 0 in such a case. Here, we're dealing with `CairoException` and its `messagePosition` field's value is 0 by default, so this call was redundant. I removed it in d80b8106f028d95796ac4a066dc0aba4640c6717